packages/devtools_app/lib/src/screens/performance/performance_model.dart

```diff
@@ -1238,77 +1243,112 @@ class FrameAnalysis {
 
   static const paintEventName = 'Paint';
 
-  /// For a single flutter frame there can be more than one build event.
-  ///
-  /// Example:
-  /// [-----------------Layout-----------------]
-  ///    [--Build--]     [----Build----]
-  List<SyncTimelineEvent> get buildEvents {
-    if (frame.timelineEventData.uiEvent == null) return <SyncTimelineEvent>[];
-    final events = _buildEvents;
-    if (events != null) return events;
-
-    _buildEvents = <SyncTimelineEvent>[];
+  static const rasterEventName = 'Raster';
 
-    final firstBuildEvent =
-        frame.timelineEventData.uiEvent!.firstChildWithCondition(
-      (event) => event.name!.caseInsensitiveEquals(buildEventName),
-    );
-    if (firstBuildEvent != null) {
-      final buildParent = firstBuildEvent.parent;
-      if (buildParent != null) {
-        _buildEvents = buildParent.children
-            .where((child) => child.name!.caseInsensitiveEquals(buildEventName))
-            .cast<SyncTimelineEvent>()
-            .toList();
-      }
-    }
-    return _buildEvents!;
-  }
+  ValueListenable<FramePhase?> get selectedPhase => _selectedPhase;
 
-  List<SyncTimelineEvent>? _buildEvents;
+  final _selectedPhase = ValueNotifier<FramePhase?>(null);
 
-  Duration get buildTime => _buildTime ??= buildEvents
-          .fold<Duration>(Duration.zero, (previous, SyncTimelineEvent event) {
-        return previous + event.time.duration;
-      });
+  void selectFramePhase(FramePhase block) {
+    _selectedPhase.value = block;
+  }
 
-  Duration? _buildTime;
+  /// Data for the build phase of [frame].
+  ///
+  /// This is drawn from all the "Build" events on the UI thread. For a single
+  /// flutter frame, there can be more than one build event, and this data may
+  /// overlap with a portion of the [layoutPhase] if "Build" timeline events are
+  /// children of the "Layout" event.
+  ///
+  /// Example:
+  /// [-----Build----][-----------------Layout-----------------]
+  ///                       [--Build--]     [----Build----]
+  late FramePhase buildPhase = _generateBuildPhase();
+
+  FramePhase _generateBuildPhase() {
+    final uiEvent = frame.timelineEventData.uiEvent;
+    if (uiEvent == null) {
+      return FramePhase.build(events: <SyncTimelineEvent>[]);
+    }
+    final buildEvents = uiEvent
+        .childrenWithCondition(
+          (event) => event.name!.caseInsensitiveEquals(buildEventName),
```

Comment:
Can `name` be null?

Reply:
it shouldn't be for any child of uiEvent, but I added null checking to be safe.